src/server/req_manager.c

```diff
@@ -2587,8 +2587,11 @@ mgr_node_unset(struct batch_request *preq)
 							reply_badattr(rc, bad, plist, preq);
 							break;
 
-						default:  req_reject(rc, 0, preq);
+						default:
+							req_reject(rc, 0, preq);
 					}
+					free(warn_nodes);
+					free(problem_nodes);
```

Comment:
I don't think we *need* this free(problem_nodes) here.  Because earlier on line 2531 we only allocate memory if numnodes > 1.  And here at line 2594 we are in the else case of numnodes > 1.
But here's a question...I also noticed that warn_nodes is allocated always...  In case someone decides to make problem_nodes look like warn_nodes and allocate always, should we leave the free(problem_nodes) there just in case?  Because it doesn't do any harm (except for the extra call to free with a NULL).  And it could help prevent leaks in the future.  What do you think?

Reply:
You are right Lisa. While this free will have no effect when problem_nodes is not allocated but I think it is not suitable to have it here for now. I think I should remove it from the code.